Extensions/DialogueTree/JsExtension.js

```diff
@@ -0,0 +1,512 @@
+ * and search for any errors.
+ *
+ * More information on https://github.com/4ian/GDevelop/blob/master/newIDE/README-extensions.md
+ */
+module.exports = {
+	createExtension: function(_, gd) {
+		const extension = new gd.PlatformExtension();
+		extension
+			.setExtensionInformation(
+				'DialogueTree',
+				_('Dialoguetree'),
+				_('Initiate dialogue trees from Yarn via Bondagejs.'),
+				'Todor Imreorov',
+				'Open source (MIT License)'
+			)
+			.setExtensionHelpPath('/all-features/dialoguetree');
+
+		extension
+			.addAction(
+				'LoadDialogueFromSceneVariable',
+				_('Load dialogue Tree from a Scene Variable'),
+				_('Load a dialogue data object - Yarn json format, stored in a scene variable. Use this command to load all the Dialogue data at the beginning of the game.'),
+				_('Load dialogue data from Scene variable _PARAM1_'),
+				_('Dialogue Tree'),
+				'JsPlatform/Extensions/yarn24.png',
+				'JsPlatform/Extensions/yarn32.png'
+			)
+			.addParameter('scenevar', _('Scene variable that holds the Yarn Json data'))
+			.getCodeExtraInformation()
+			.setIncludeFile('Extensions/DialogueTree/dialoguetools.js')
+			.addIncludeFile('Extensions/DialogueTree/bondage.min.js')
+			.setFunctionName('gdjs.dialogueTree.loadFromSceneVariable');
+
+		extension
+			.addAction(
+				'LoadDialogueFromJsonFile',
+				_('Load dialogue Tree from a Json File'),
+				_('Load a dialogue data object - Yarn json format, stored in a Json file. Use this command to load all the Dialogue data at the beginning of the game.'),
+				_('Load dialogue data from json file _PARAM1_'),
+				_('Dialogue Tree'),
+				'JsPlatform/Extensions/yarn24.png',
+				'JsPlatform/Extensions/yarn32.png'
+			)
+			.addCodeOnlyParameter('currentScene', '')
+			.addParameter('jsonResource', _('Json file that holds the Yarn Json data'))
+			.getCodeExtraInformation()
+			.setIncludeFile('Extensions/DialogueTree/dialoguetools.js')
+			.addIncludeFile('Extensions/DialogueTree/bondage.min.js')
+			.setFunctionName('gdjs.dialogueTree.loadFromJsonFile');
+
+		extension
+			.addAction(
+				'StarDialogueFromBranch',
+				_('Start dialogue from branch'),
+				_('Start dialogue from branch. Use this to initiate the dialogue from a specified branch.'),
+				_('Start dialogue from branch _PARAM0_'),
+				_('Dialogue Tree'),
+				'JsPlatform/Extensions/yarn24.png',
+				'JsPlatform/Extensions/yarn32.png'
+			)
+			.addParameter('string', _('Dialogue branch'))
+			.getCodeExtraInformation()
+			.setFunctionName('gdjs.dialogueTree.startFrom');
+
+		extension
+			.addAction(
+				'GoToNextLine',
+				_('Go to the Next dialogue Line'),
+				_('Go to the Next dialogue line. Use this to advance to the next dialogue line  when the player presses a button.'),
+				_('Go to the Next dialogue Line'),
+				_('Dialogue Tree'),
+				'JsPlatform/Extensions/yarn24.png',
+				'JsPlatform/Extensions/yarn32.png'
+			)
+			.getCodeExtraInformation()
+			.setFunctionName('gdjs.dialogueTree.advanceDialogue');
```

Comment:
This should be called "goToNextDialogueLine" if you want to be consistent with the action name? Either do that or keep "advance" everywhere